### PR TITLE
Add support for const schemas, select boxes for oneOf schemas

### PR DESCRIPTION
**Pull request by helendurranttw**
## ❓ Context  
We would like to be able to support a native JSON schema approach for select boxes
```
{
  title: "Favourite number?",
  oneOf: [
    { title: "One", const: 1 }
    { title: "Two", const: 2 }
  ]
}
```

## 🚀 Changes 
This changes oneOf schemas to display as select boxes when there are more options, and support const schemas which broadcast their value when chosen.

## 💬 Considerations 
n/a

## ✅ Checklist

- [x] All changes are covered by tests
- [x] The changes are covered in docs
- [x] The commits follow conventional commits standards

### CODE REVIEW

File: packages/dynamic-flows/src/common/validation/type-validators/spec.js [hunk 1]
```diff
@@ -94,4 +103,30 @@ describe('Given a library for validating data types', () => {
...
+      expect(isNull(0)).toBe(false);
+      expect(isNull(false)).toBe(false);
+      expect(isNull([])).toBe(false);
+      expect(isNull({})).toBe(false);
+      expect(isNull(undefined)).toBe(false);
+    });
+  });
+
+  describe('when validating an undefined', () => {
+    it('should return true when the value is null', () => {
```
Comment (helendurranttw) [inline packages/dynamic-flows/src/common/validation/type-validators/spec.js:121]: Typo: undefined, not null

File: packages/dynamic-flows/src/common/validation/type-validators/spec.js [hunk 2]
```diff
@@ -94,4 +103,30 @@ describe('Given a library for validating data types', () => {
...
+      expect(isUndefined(undefined)).toBe(true);
+    });
+    it('should return false when the value is not null', () => {
```
Comment (helendurranttw) [inline packages/dynamic-flows/src/common/validation/type-validators/spec.js:124]: Typo: undefined, not null

File: packages/dynamic-flows/src/formControl/FormControl.js [hunk 3]
```diff
@@ -232,6 +232,14 @@ export default class FormControl extends PureComponent {
     return selectedOption;
   };
 
+  mapOption = (option) => {
+    return {
+      ...option,
+      disabled: option.disabled || this.props.disabled,
+      readOnly: this.props.readOnly,
```
Comment (helendurranttw) [inline packages/dynamic-flows/src/formControl/FormControl.js:239]: I guess it doesn't make sense for radioButtons, but should we also prefer `option.readOnly` over `props.readOnly` here?

File: packages/dynamic-flows/src/jsonSchemaForm/validation/valid-model/spec.js [hunk 4]
```diff
@@ -76,6 +76,63 @@ describe('Given a library for returning the valid parts of a model based on a sc
...
+      it('should return the original model', () => {
+        expect(result).toBe(1);
+      });
+    });
+
+    describe('if the model is not part of the enum', () => {
+      beforeEach(() => {
+        result = getValidModelParts(3, schema);
+      });
+      it('should return undefined', () => {
```
Comment (helendurranttw) [inline packages/dynamic-flows/src/jsonSchemaForm/validation/valid-model/spec.js:97]: Undefined, or null?

File: packages/dynamic-flows/src/jsonSchemaForm/validation/valid-model/spec.js [hunk 5]
```diff
@@ -76,6 +76,63 @@ describe('Given a library for returning the valid parts of a model based on a sc
...
+        expect(result).toBe(1);
+      });
+    });
+
+    describe('if the model does not match the const', () => {
+      beforeEach(() => {
+        result = getValidModelParts(2, schema);
+      });
+      it('should return undefined', () => {
+        expect(result).toBeNull();
```
Comment (helendurranttw) [inline packages/dynamic-flows/src/jsonSchemaForm/validation/valid-model/spec.js:122]: Undefined, or null?

File: packages/dynamic-flows/src/formControl/FormControl.js (hunk 3 above)
Comment (StevePole) [reply packages/dynamic-flows/src/formControl/FormControl.js:239]: I don't think there is a concept of a readonly option, as options are never editable.  I presume that by passing it here, we're actually just disabling the option as part of broader control.  However, I was really just preserving the existing behaviour.

File: packages/dynamic-flows/src/jsonSchemaForm/validation/valid-model/spec.js (hunk 4 above)
Comment (StevePole) [reply packages/dynamic-flows/src/jsonSchemaForm/validation/valid-model/spec.js:97]: null, good spot.

**Pull request merged**